### ISSUE TITLE
[battery_manager] Unable to access module without superuser permission 500 error

### ISSUE THREAD

**Issue by ridz1208** [aces/Loris#7798]
**Describe the bug**
When trying to access the module with tthe regular battery_manager permissions (without superuser) there is a 500 error thrown.

```
[Fri Nov 05 15:29:54.800442 2021] [php:error] [pid 1347] [client 192.168.122.1:41504] PHP Fatal error:  Uncaught TypeError: User::hasCenter(): Argument #1 ($center_id) must be of type CenterID, int given, called in /var/www/Loris/src/Data/Filters/UserSiteMatch.php on line 48 and defined in /var/www/Loris/php/libraries/User.class.inc:433\nStack trace:\n#0 /var/www/Loris/src/Data/Filters/UserSiteMatch.php(48): User->hasCenter()\n#1 /var/www/Loris/src/Data/ProvisionerInstance.php(129): LORIS\\Data\\Filters\\UserSiteMatch->filter()\n#2 [internal function]: LORIS\\Data\\ProvisionerInstance->LORIS\\Data\\{closure}()\n#3 [internal function]: CallbackFilterIterator->accept()\n#4 /var/www/Loris/src/Data/Table.php(83): FilterIterator->rewind()\n#5 /var/www/Loris/modules/battery_manager/php/testendpoint.class.inc(254): LORIS\\Data\\Table->toArray()\n#6 /var/www/Loris/modules/battery_manager/php/testendpoint.class.inc(109): LORIS\\battery_manager\\TestEndpoint->_getInstances()\n#7 /var/www/Loris/modules/battery_manager/php/testendpoint.class.inc(76): LORIS\\battery_manager\\TestEndpoint->handle()\n#8 /var/www/Loris/php/libraries/Module.class.inc(398): LORIS\\battery_manager\\TestEndpoint->process()\n#9 /var/www/Loris/src/Middleware/ResponseGenerator.php(50): Module->handle()\n#10 /var/www/Loris/src/Middleware/AuthMiddleware.php(63): LORIS\\Middleware\\ResponseGenerator->process()\n#11 /var/www/Loris/src/Router/ModuleRouter.php(74): LORIS\\Middleware\\AuthMiddleware->process()\n#12 /var/www/Loris/src/Middleware/ExceptionHandlingMiddleware.php(39): LORIS\\Router\\ModuleRouter->handle()\n#13 /var/www/Loris/src/Router/BaseRouter.php(126): LORIS\\Middleware\\ExceptionHandlingMiddleware->process()\n#14 /var/www/Loris/src/Middleware/ResponseGenerator.php(50): LORIS\\Router\\BaseRouter->handle()\n#15 /var/www/Loris/src/Middleware/ContentLength.php(52): LORIS\\Middleware\\ResponseGenerator->process()\n#16 /var/www/Loris/htdocs/index.php(55): LORIS\\Middleware\\ContentLength->process()\n#17 {main}\n  thrown in /var/www/Loris/php/libraries/User.class.inc on line 433, referer: https://test-loris-dev.loris.ca/battery_manager/
```

**Comment by driusan** [comment]
I looked into the issue tracker for #7799 and there seems to be some code paths that phan wasn't able to track down the type incompatibility because the docblock signature is wrong. In the case of the issue tracker the docblock for the `IssueRow` says it takes an array of unspecified types. This is then used by getCenterIDs (for the MultiSiteHaver interface) which must be a more specific `\CenterID[]` type. At the call site of the constructor, it's incorrectly passed an array of ints in IssueRowProvisioner, but since the constructor was an unspecified array type it doesn't get flagged.

I assume the battery_manager is similar. @ridz1208 do you have a list of modules where this was encountered? I think we should reopen #7799 and make separate issues for any that have the problem. They each need to be fixed individually.

**Comment by ridz1208** [comment]
@driusan I think you ended up looking for the separate instances.

#7799 was reopened

Also related to #7764 I think

**Comment by driusan** [comment]
I did a grep in the code and those are the only 3 modules that implement MultiSiteHaver, so they should be the only 3 places that have the problem.

**Comment by driusan** [comment]
Fixed by #7815
